aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -461,7 +461,7 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cuda(const Tensor& grad_o
   // save_mean and save_invstd, so it needs recalculated.
   const auto acc_type = at::toAccumulateType(input.scalar_type(), /*is_cuda=*/true);
   Tensor mean;
-  if (save_mean->defined()) {
+  if (save_mean->defined() && save_mean->numel() != 0) {
```

Comment:
Given the modification on the caller, should this assert that save_mean is always defined?
Same for save_invstd below

Reply:
I think so. I can try adding that and run it through CI to see if we upset anything.